astro/src/content/articles/authentication/bff.mdx

```diff
@@ -0,0 +1,319 @@
+- **Auth0:** Agrees that [storing tokens in the browser is dangerous](https://auth0.com/docs/secure/security-guidance/data-security/token-storage), and recommends storing them in RAM, not local storage. Auth0 doesn't provide a hosted backend. The [Auth0 SPA SDK](https://auth0.com/docs/quickstart/spa/vanillajs/interactive) sends the access token directly to the browser.
+- **Cloudentity:** [Recommends using an OAuth proxy](https://cloudentity.com/developers/blog/adding-oauth-proxy-bff-component-to-spa/), but doesn't appear to offer one as part of its service.
+- **Duende:** Offers a self-hosted [ASP.NET OAuth Proxy package](https://blog.duendesoftware.com/posts/20210326_bff/) as part of its IdentityServer.
+
+Based on their documentation, Okta, Ory, and Keycloak don't appear to offer built-in OAuth proxies.
+
+## Summary
+
+This article covers a lot, but the recommendations are straightforward:
+
+- **Use `HttpOnly` cookies in the browser.** Never store secrets in local storage or session storage.
+- **Add a server for any app that uses OAuth**, even mobile and desktop apps if possible, so tokens are never sent to the browser. If your app is currently serverless, add an OAuth proxy to make it safer.
+- **Choose a simple OAuth proxy approach**, such as using the FusionAuth hosted backend or Go Auth2 Proxy, or building on the Passport.js Node sample in this article. (Other languages have OAuth wrappers, too, so you won't have to write your own protocol code.)
+- **Avoid malicious dependencies** by being careful about the packages you build with.
+- **Consider [multifactor authentication](/docs/lifecycle/authenticate-users/multi-factor-authentication)** for dangerous actions like buying, selling, or accessing private information.
+
+## Further Reading
+
+- [Phil Calçado's 2015 post explaining the BFF pattern at SoundCloud](https://philcalcado.com/2015/09/18/the_back_end_for_front_end_pattern_bff.html)
+- [Dropbox OAuth documentation](https://www.dropbox.com/developers/documentation/http/documentation)
+- [What OAuth tokens are used for](/blog/what-happens-after-oauth-authentication-event)
+- [Token storage locations](/articles/oauth/oauth-token-storage)
+- [Passport.js](https://www.passportjs.org/) and the [OAuth 2.0 extension](https://github.com/jaredhanson/passport-oauth2) for JavaScript servers
+- [FusionAuth React tutorial for the hosted backend](/docs/quickstarts/quickstart-javascript-react-web)
+- [FusionAuth hosted backend API](/docs/apis/hosted-backend)
+- [FusionAuth OAuth scopes](/docs/lifecycle/authenticate-users/oauth/scopes)
+- [Node `http-server` package](https://www.npmjs.com/package/http-server)
+- [Go OAuth2 Proxy](https://oauth2-proxy.github.io/oauth2-proxy)
+- [Multifactor authentication in FusionAuth](/docs/lifecycle/authenticate-users/multi-factor-authentication)
+
```

Comment:
Add the following Further Reading resource:

- [OAuth 2.0 for Browser-Based Applications](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-backend-for-frontend-bff): [Backend For Frontend](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-backend-for-frontend-bff) and [Token-Mediating Backend](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-token-mediating-backend)